app/controllers/homepage_controller.rb

```diff
@@ -37,20 +37,38 @@ def index
 
     listings_per_page = APP_CONFIG.grid_listings_limit
 
+    filter_params = {}
+
+    # :transaction_type was renamed to :order_type
+    # Support both URLs for a while
+    # This can be removeds soon (June 2015)
+    params[:order_type] ||= params[:transaction_type]
+    params[:transaction_type] = nil
+    order_type_params = params[:order_type]
+
+    all_shapes = shapes.get(community_id: @current_community.id)[:data]
+    selected_shape = all_shapes.find { |s| s[:name] == order_type_params }
+
+    filter_params[:listing_shape] = Maybe(selected_shape)[:id].or_else(nil)
+
+    compact_filter_params = HashUtils.compact(filter_params)
```

Comment:
This action is overly long already. Looks like there's couple of easy extract methods to parse parameters, to get shapes, and to build filter params from shapes.